File: backend/scripts/gerar-base-json.js
// Gera dist/base.json a partir do index.html do HomeoVet (fonte de verdade)
const fs = require('fs');
const path = require('path');

const SRC = path.join(__dirname, '..', '..', '..', 'homeovet', 'index.html');
const DEST = path.join(__dirname, '..', '..', 'dist', 'base.json');

if (!fs.existsSync(SRC)) {
  console.error('index.html não encontrado em', SRC);
  process.exit(1);
}

const html = fs.readFileSync(SRC, 'utf-8');
const m = html.match(/const BASE = (\{.*\});\r?\n/s);
if (!m) {
  console.error('BASE não encontrada no index.html');
  process.exit(1);
}

const base = JSON.parse(m[1]);
fs.mkdirSync(path.dirname(DEST), { recursive: true });
fs.writeFileSync(DEST, JSON.stringify(base, null, 2), 'utf-8');
console.log(`✅ base.json gerado: ${base.medicamentos.length} medicamentos, ${base.evidencias_cientificas.length} evidências, ${base.regulamentacao_brasil.length} regulamentações, ${Object.keys(base.glossario).length} termos`);

File: backend/scripts/gerar-base-json.py
# -*- coding: utf-8 -*-
"""Gera dist/base.json a partir do index.html do HomeoVet (fonte de verdade)."""
import re, json, os, sys

SRC = r"C:\Users\JEANPC\homeovet\index.html"
DEST = r"C:\Users\JEANPC\homeovet-app\dist\base.json"

html = open(SRC, encoding="utf-8").read()
m = re.search(r"const BASE = (\{.*?\});\n", html, re.S)
if not m:
    print("ERRO: BASE não encontrada no index.html")
    sys.exit(1)

base = json.loads(m.group(1))
os.makedirs(os.path.dirname(DEST), exist_ok=True)
with open(DEST, "w", encoding="utf-8") as f:
    json.dump(base, f, ensure_ascii=False, indent=2)

print(f"OK: base.json gerado — {len(base['medicamentos'])} medicamentos, "
      f"{len(base['evidencias_cientificas'])} evidências, "
      f"{len(base['regulamentacao_brasil'])} regulamentações, "
      f"{len(base['glossario'])} termos")
